packages/devtools_app/test/vm_developer/object_inspector/vm_developer_common_widgets_test.dart

```diff
@@ -0,0 +1,218 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app/devtools_app.dart';
+import 'package:devtools_app/src/screens/vm_developer/vm_developer_common_widgets.dart';
+import 'package:devtools_test/devtools_test.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../vm_developer_test_utils.dart';
+
+void main() {
+  const windowSize = Size(4000.0, 4000.0);
+
+  late InstanceRef requestedSize;
+
+  int size = 0;
+
+  late void Function() requestFunction;
+
+  bool fetchRetainingPath = false;
+
+  late ValueNotifier<RetainingPath?> retainingPathNotifier;
+
+  late void Function(bool) fetchRetainingPathFunc;
+
+  bool fetchInboundRefs = false;
+
+  late ValueNotifier<InboundReferences?> inboundRefsNotifier;
+
+  late void Function(bool) fetchinboundRefsFunc;
+
+  setUp(() {
+    setGlobal(IdeTheme, IdeTheme());
+
+    final json = testInstance.toJson();
+    requestedSize = Instance.parse(json)!;
+    requestedSize.valueAsString = '1024';
+
+    requestFunction = () {
```

Comment:
Any particular reason we're recreating these closures for each test?

Reply:
Because we need to pass the closure as an argument to the widgets. I think I can use a mock object, but still have to pass the closure to the widget to know what data it has to request.